pilot/pkg/serviceregistry/serviceentry/servicediscovery.go

```diff
@@ -438,8 +438,15 @@ type servicesWithEntry struct {
 	services []*model.Service
 }
 
+// Resync EDS will do a full EDS update. This is needed for some tests where we have many configs loaded without calling
+// the config handlers.
+// This should probably not be used in production code.
+func (s *ServiceEntryStore) ResyncEDS() {
+	s.edsUpdate(nil, true)
+}
+
 // edsUpdate triggers an EDS update for the given instances
-func (s *ServiceEntryStore) edsUpdate(instances []*model.ServiceInstance) {
+func (s *ServiceEntryStore) edsUpdate(instances []*model.ServiceInstance, full bool) {
```

Comment:
Nit: Instead of changing signature of this, should we pass `s.instances` in `ResyncEDS`

Reply:
I am not sure how that avoids changing the signature?

Reply:
I am thinking the idea here is to look at all instances i.e. `s.instances` when full is true in resync?- instead of updated instances that we normally pass during eds update? Then if we pass `s.instances` the `keys` that we construct will have all - right?  Am I missing some thing?
Anyway - it was just a nit - not a concern for me

Reply:
I get what you mean - SGTM, I made the change